src/include/main/client_context.h

```diff
@@ -1,38 +1,55 @@
 #pragma once
 
 #include <atomic>
+#include <chrono>
 #include <cstdint>
+#include <memory>
 
 #include "common/api.h"
+#include "main/kuzu_fwd.h"
 
 namespace kuzu {
 namespace main {
 
+struct ActiveQuery {
+    explicit ActiveQuery();
+
+    std::chrono::time_point<std::chrono::high_resolution_clock> startTime;
+    std::atomic<bool> interrupted;
+};
+
 /**
  * @brief Contain client side configuration. We make profiler associated per query, so profiler is
  * not maintained in client context.
  */
 class ClientContext {
     friend class Connection;
+    friend class testing::TinySnbDDLTest;
+    friend class testing::TinySnbCopyCSVTransactionTest;
 
 public:
-    /**
-     * @brief Constructs the ClientContext object.
-     */
-    KUZU_API explicit ClientContext();
-    /**
-     * @brief Deconstructs the ClientContext object.
-     */
-    KUZU_API ~ClientContext() = default;
-
-    /**
-     * @brief Returns whether the current query is interrupted or not.
-     */
-    KUZU_API bool isInterrupted() const { return interrupted; }
+    explicit ClientContext();
```

Comment:
I guess we are not exposing clientContext so maybe we don't need to pay much attention to the public interface here.

Also, I wouldn't start timing in constructor. Instead I would use a startTiming() to explicitly control when to start. 

Also refactor this with how we time our execution time. I think we already have a timer library.

Reply:
reused the timer library and implemented startTiming()